x/xdocker/xarchive/archive.go

```diff
@@ -0,0 +1,14 @@
+package xarchive
+
+import (
+	"io"
+
+	"github.com/docker/docker/pkg/archive"
+)
+
+// GzippedTar creates an archive from the directory at path compressed with gzip.
+func GzippedTar(path string) (io.Reader, error) {
```

Comment:
This function is not used anywhere. Maybe a problem with the merge?

The codebase is still using `TarWithOptions`

Reply:
i requested this one, will be used by tests

Reply:
Why not using it in the codebase instead of `TarWithOptions`?

Reply:
yes we should

Reply:
As @ilgooz pointed out, the function will be used

Reply:
So, please update the codebase to use this `GzippedTar` function or at least create some issue to not forget. Otherwise, I worry that we will end up with a lot of function in `x` packages that no other function.

Here are some place that use `TarWithOptions`:

https://github.com/mesg-foundation/core/blob/8711a71037209288d39287805c7653f7bb043da9/container/build.go#L26-L29

https://github.com/mesg-foundation/core/blob/8711a71037209288d39287805c7653f7bb043da9/mesg/deploy_test.go#L30

https://github.com/mesg-foundation/core/blob/8711a71037209288d39287805c7653f7bb043da9/mesg/deploy_test.go#L77

Reply:
Same with git clone, it's another branch

Reply:
Done